libcst/metadata/scope_provider.py

```diff
@@ -740,7 +747,10 @@ def _handle_string_annotation(
 
     def visit_Subscript(self, node: cst.Subscript) -> Optional[bool]:
         qnames = self.scope.get_qualified_names_for(node.value)
-        if any(qn.name.startswith(("typing.", "typing_extensions.")) for qn in qnames):
+        if (
+            any(qn.name.startswith(("typing.", "typing_extensions.")) for qn in qnames)
+            and not self.__ignore_annotation
+        ):
```

Comment:
We still want to do the work in `_handle_string_annotation` for these cases, with the only difference being `Access` nodes should have `is_annotation=False`.